GNUmakefile

```diff
@@ -43,7 +43,12 @@ vet:
 		go get golang.org/x/tools/cmd/vet; \
 	fi
 	@echo "--> Running go tool vet $(VETARGS) ."
-	@go tool vet $(VETARGS) . ; if [ $$? -eq 1 ]; then \
+	@go list ./... \
+		| grep -v ^github.com/hashicorp/consul/vendor/ \
+		| cut -d '/' -f 4- \
+		| xargs -n1 \
+			go tool vet $(VETARGS) ;\
+	if [ $$? -eq 1 ]; then \
```

Comment:
@sean- Actually I think this might break with the way `xargs` maps return codes. This should probably be changed to a != 0 check, and we could `set -o pipefail` here to make sure we know if any of the other commands die.